deploy/dynamo/sdk/src/dynamo/sdk/cli/serving.py

```diff
@@ -127,6 +128,29 @@ def _get_server_socket(
 _DYNAMO_WORKER_SCRIPT = "dynamo.sdk.cli.serve_dynamo"
 
 
+def maybe_get_slurm_mpirun_command(service_name, args):
+    # HACK: WAR issues with running MPI_Spawn in TRTLLM internals
+    #       while in a Slurm environment.
+    # FIXME: Find a better solution for running TRTLLM workers with `dynamo serve`
+    is_slurm = os.environ.get("SLURM_NODELIST") is not None
+    if is_slurm and "TensorRTLLM" in service_name:
+        logger.info(
+            "Slurm environment detected: Pre-pending `mpirun` to TRTLLM workers."
+        )
+        args = [
+            "-np",
+            "1",
+            "--allow-run-as-root",
```

Comment:
I am very skeptical about this. I know this is how we are currently able to run trtllm on slurm. But I am not sure about the complete implications of `--allow-run-as-root` option. Is this advisable on a production environment. Should we add a warning in our TRTLLM example describing that dynamo serve will use `--allow-run-as-root` to launch TRTLLM workers on slurm?